test/autotest.py

```
@@ -1760,6 +1805,35 @@ def _disabled_reasons(cls, test: TestSpec) -> List[str]:
+        # Trial-run instead of duplicating mmap-noreserve.c's REGION_BYTES:
+        # it prints READY on success, or fails fast via perror("mmap").
+        binary = ROOT / "test" / "mmap-noreserve"
+        if not binary.exists():
+            return None  # _command_executable_reasons() will report this
+        try:
+            original = resource.getrlimit(resource.RLIMIT_AS)
+            resource.setrlimit(resource.RLIMIT_AS, (original[1], original[1]))
+        except (ValueError, OSError):
+            return None  # can't tell; let the test attempt to run
+        try:
+            proc = subprocess.Popen([str(binary)], stdout=subprocess.PIPE,
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)